### ISSUE TITLE
Follow up on ECAL Emulator Update (PR #33349)

### ISSUE THREAD

**Issue by silviodonato** [cms-sw/cmssw#33425]
@atishelmanch  @thomreis 
I have two comments about #33349

1) I see there are many `std::cout`. `cout` is not thread-safe, so it needs to be replaced with LogVerbatim or LogInfo or something similar (see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger )

2) There are a strong usage of `catch ( ... )`.  You should not catch the exception https://cms-sw.github.io/cms_coding_rules.html . 
Some of them were already present in the code (@ggovi I don't know if this is somehow allowed for SQLException).

> In general, do not catch exceptions – leave them to the Framework (see Exception Guidelines).

**Comment by silviodonato** [comment]
assign l1,db

**Comment by thomreis** [comment]
As far as I can see all the unprotected `cout` and `catch` are in the `CondTools/Ecal` and the `OnlineDB` packages. I believe that the code in there handles writing conditions to the Online DB and does only run single threaded.
One can find `cout` statements in many other files in that package as well. 

Also the `catch` statements are seen at many places in OnlineDB, dating back to at least CMSSW_5_0_0.

For the odd weights code we duplicated the existing even weights code that contained the `cout` and `catch` already.

The remaining `cout` in the `SimCalorimetry` should be all protected by a boolean that is only manually set to true for debugging. We can change those to `LogVerbatim` or `LogInfo` if necessary.

**Comment by thomreis** [comment]
Hi @ggovi can you comment on the usage of `cout` and `catch` in these packages please. Is this OK?

**Comment by thomreis** [comment]
Hi @ggovi do you have any comments on this issue?

**Comment by ggovi** [comment]
I'm sorry for the latency. In my opinion, the Std::cout  printouts could be accepted only in a test executable. In all the Analyzer code, instead, std::cout should not be used. On the other hand, you have a significant fraction of your messages already implemented with LogInfo or LogDebug. I really think you should be uniform and coherent on this.
Concerning the catch (...), I see that is limited to two test executable. In general, it indicate as not well understood troubleshoot condition, which in concrete needs to be investigated.  If instead you are using it 'just in case', I would rather remove it.

**Comment by thomreis** [comment]
We can replace the cout statements with LogDebug or LogInfo/LogVerbatim to be consistent, yes.

For the catch(...) statements I count 46 occurrences. Some are catching std::exceptions and are just printing a message and the exception itself. Others do the same but also setting some variable. Most others catch an SQLException and rethrow it as a std::runtime_error. And then there are the ones in the test executables.

For the ones catching the SQLExceptions should they rather rethrow cms::exceptions?

**Comment by ggovi** [comment]
My comment was actually limited to the actual **catch(...)** , that are only two.

**Comment by thomreis** [comment]
Ah yes, sorry. I see what you mean now.
Can the others be kept as they are then?

**Comment by ggovi** [comment]
Well you have a recommendation from the release coordinator... Please check the guidelines and decide based on your needs - in same case I've seen you simply rethrow...

**Comment by thomreis** [comment]
Yes we duplicated the code that has been there for ages for the integration of the second set of weights.

**Comment by tvami** [comment]
Hi @thomreis has this issue ever been resolved?

**Comment by thomreis** [comment]
No, this is still on the to do list. @atishelmanch @valsdav

**Comment by thomreis** [comment]
@tvami this issue should be handled in #35557.

**Comment by tvami** [comment]
assign db

**Comment by thomreis** [comment]
Hi @tvami with #35557 merged I think we can close this issue.
